packages/pigeon/lib/kotlin_generator.dart

```diff
@@ -839,6 +1099,615 @@ class KotlinGenerator extends StructuredGenerator<KotlinOptions> {
       });
     });
   }
+
+  void _writeProxyApiRegistrar(
+    Indent indent, {
+    required Iterable<AstProxyApi> allProxyApis,
+  }) {
+    const String registrarName = '${classNamePrefix}ProxyApiRegistrar';
+
+    indent.format(
+      '/**\n'
+      ' * Provides implementations for each ProxyApi implementation and provides access to resources\n'
+      ' * needed by any implementation.\n'
+      ' */',
+    );
+    indent.writeScoped(
+      'abstract class $registrarName(val binaryMessenger: BinaryMessenger) {',
+      '}',
+      () {
+        indent.writeln('val instanceManager: PigeonInstanceManager');
+        indent.format(
+          'private var _codec: StandardMessageCodec? = null\n'
```

Comment:
This will need to change to use the new codec - once the codec pr lands of course